kuwala/core/canvas/src/components/Modals/ExportBlockConfig/ExportBlockConfigModal.js

```diff
@@ -0,0 +1,205 @@
+    const [ submitData, setSubmitData ] = useState(null);
+    const initNodeName = useCallback(() => {
+        if (selectedElement && selectedElement.type === EXPORT_BLOCK) {
+            setExportBlockName(selectedElement.data.exportBlock.name)
+        } else {
+            setExportBlockName(undefined);
+        }
+    }, [selectedElement]);
+
+    useEffect( () => {
+        initNodeName()
+    }, [initNodeName])
+
+    const onNameChange = (event) => {
+        setExportBlockName(event.target.value);
+    }
+
+    const FormikWrapper = ({children}) => {
+        const mapParameter = (param) => {
+            let value = '';
+
+            if (param.id === 'aggregated_columns') {
+                value = [{ column: null, aggregation: null}]
+            } else if (param.type === 'list[text]') {
+                value = ['']
+            }
+
+            return { ...param, value }
+        }
+        let params = selectedElement.data.exportCatalogItem.macroParameters.map(mapParameter);
+
+        if(selectedElement.data.exportBlock.isConfigured) {
+            params = selectedElement.data.exportBlock.macroParameters;
+        }
+
+        let materializeTable = selectedElement.data.exportBlock.materializeTable;
+        let exBlockName = exportBlockName;
+
+        if(submitData !== null) {
+            params = submitData.parameters;
+            materializeTable = submitData.materializeTable;
+            exBlockName = submitData.exportBlockName;
+        }
+
+        return (
+            <Formik
+                initialValues={{
+                    parameters: params,
+                    materializeTable: materializeTable,
+                    exportBlockName: exBlockName,
+                }}
+                onSubmit={async (values)=>{
+                    setSubmitData(values);
+                    await upsertExportBlock({ values });
+                    setSubmitData(null);
+                }}
+                children={() => (
+                    <>
+                        {children}
+                    </>
+                )}
+            />
+        )
+    }
+
+    const toggleConfigModalWrapper = () => {
+        toggleExportConfigModal();
+        setSubmitData(null);
+    }
+
+    const upsertExportBlock = async ({values}) => {
+        setIsExportBlockSaveLoading(true);
+        const connectedElements = getElementByIds(elements, selectedElement.data.exportBlock.connectedSourceNodeIds);
+        const connectedBlocks = connectedElements.map((el) => getEntityElementEntityBlockId(el));
+        const mapParameter = (param) => {
+            let value = param.value;
+
+            if (param.type === 'date') {
+                value = moment(param.value).format('YYYY-MM-DD')
+            } else if (param.id === 'aggregated_columns') {
+                value = param.value.map(({ column, aggregation }) => `${column}KUWALA_AGG${aggregation}`);
+            }
+
+            return { ...param, value };
+        }
```

Comment:
This is a duplicated code fragment. Please refactor it to a utils function.
![image](https://user-images.githubusercontent.com/28758040/174645462-9d8b7834-ec8e-44d7-9ba4-8e9db6ead424.png)

Reply:
Created a new common function for macro parameter operations at `MacroParameterUtils.js`